### DISCUSSION TITLE
issue: defaultValues of required number inputs ignored on form submit if not rendered first

### DISCUSSION THREAD

**Discussion by elieven** [react-hook-form/react-hook-form#9568] [Q&A]
### Version Number

7.40.0

### Codesandbox/Expo snack

https://codesandbox.io/s/react-hook-form-tabs-problems-01-required-numbers-kmy5jp?file=%2Fsrc%2FApp.tsx

### Steps to reproduce

1. Go to https://codesandbox.io/s/react-hook-form-tabs-problems-01-required-numbers-kmy5jp?file=%2Fsrc%2FApp.tsx
2. Click on submit (without clicking on any other tabs)
3. Look at console log
4. You should see `errors {prop2: Object, prop3: Object}`

### Expected behaviour

Form should submit without any errors since the default values have been provided for all required inputs. The exact same thing but with "text" inputs works as expected and submits the form with all data from `defaultValues` and no errors as can be seen in this sandbox: https://codesandbox.io/s/react-hook-form-tabs-problems-01-required-strings-9btufh?file=%2Fsrc%2FApp.tsx.

### What browsers are you seeing the problem on?

Firefox, Chrome, Edge

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by elieven** [comment]
It's particularly strange because if you click on the other tabs (cause the other inputs to render too) it works. But if all the inputs have not been rendered at some point it does not work. I'm talking about number inputs only since i tested number and text and of those the number ones caused the issues. I have not tested date or any other type of input to see if those have the same problem too.

**Comment by bluebill1049** [comment]
Never it's not to do with render, i am looking into this issue now.

**Reply by bluebill1049** [reply]
https://github.com/react-hook-form/react-hook-form/issues/9571

**Reply by bluebill1049** [reply]
I can confirm the issue is due to input is not been registered, looking for a solution to work around this.

**Reply by JoshMart** [reply]
@bluebill1049 It would be helpful to not close an issue when it's reported, move it to discussion, create a new issue for the closed issue, and link back to the original issue from the new issue. There's lots of history and discussion about the bug scattered in 3 different places.

**Comment by leapful** [comment]
@elieven @bluebill1049 

A workaround is to render all tab contents and hide it using CSS.

Working demo: https://codesandbox.io/s/react-hook-form-tabs-problems-01-required-numbers-forked-tzidgk

**Reply by bluebill1049** [reply]
Thanks @leapful i have patched it will release it this week 🙏 👍

**Reply by elieven** [reply]
@leapful I was not aware of this `[data-headlessui-state=""]` thing (_never really looked too deep into headless UI tabs_) quite an interesting trick. For my use case it was _probably_ better to not render all he fields.
